src/FSharpVSPowerTools.Core/Utils.fs

```diff
@@ -545,14 +696,18 @@ module Dict =
 [<RequireQualifiedAccess>]
 [<CompilationRepresentation(CompilationRepresentationFlags.ModuleSuffix)>]
 module String =
+
+    let inline toCharArray (str:string) = str.ToCharArray()
+
     let lowerCaseFirstChar (str: string) =
-        match str with
-        | null -> null
-        | str ->
-            match str.ToCharArray() |> Array.toList with
-            | [] -> str
-            | h :: t when Char.IsUpper h -> String (Char.ToLower h :: t |> List.toArray)
-            | _ -> str
+        if isNull str then null else 
+        let strArr = toCharArray str
```

Comment:
what about checking if string can be returned as is before doing array allocation?
```fsharp
if String.IsNullOrEmpty str || Char.IsLower(str, 0) then str
else 
// do the rest 
```